ignite/contrib/engines/common.py

```diff
@@ -98,28 +105,34 @@ def _setup_common_training_handlers(
     save_every_iters=1000,
     output_path=None,
     lr_scheduler=None,
-    with_gpu_stats=True,
+    with_gpu_stats=False,
     output_names=None,
     with_pbars=True,
     with_pbar_on_iters=True,
     log_every_iters=100,
-    device="cuda",
+    stop_on_nan=True,
 ):
-    trainer.add_event_handler(Events.ITERATION_COMPLETED, TerminateOnNan())
+    if stop_on_nan:
+        trainer.add_event_handler(Events.ITERATION_COMPLETED, TerminateOnNan())
 
     if lr_scheduler is not None:
         if isinstance(lr_scheduler, torch.optim.lr_scheduler._LRScheduler):
             trainer.add_event_handler(Events.ITERATION_COMPLETED, lambda engine: lr_scheduler.step())
+        elif isinstance(lr_scheduler, LRScheduler):
+            trainer.add_event_handler(Events.ITERATION_COMPLETED, lr_scheduler)
         else:
             trainer.add_event_handler(Events.ITERATION_STARTED, lr_scheduler)
 
-    trainer.add_event_handler(Events.EPOCH_COMPLETED, empty_cuda_cache)
+    if torch.cuda.is_available():
```

Comment:
It seems like we should let a user opt into this. Clearing the cache can mean huge performance hits in some situations.